src/definitions/behaviors/form.js

```diff
@@ -444,15 +444,11 @@ $.fn.form = function(parameters) {
                 $fieldGroup     = $field.closest($group),
                 validationRules = module.get.validation($field)
               ;
-              if( $fieldGroup.hasClass(className.error) ) {
+              if(validationRules && (settings.on == 'blur' || ( $fieldGroup.hasClass(className.error) && settings.revalidate) )) {
                 module.debug('Revalidating field', $field, validationRules);
-                if(validationRules) {
-                  module.validate.field( validationRules );
-                }
-              }
-              else if(settings.on == 'blur') {
-                if(validationRules) {
-                  module.validate.field( validationRules );
+                module.validate.field( validationRules );
+                if(!settings.inline) {
```

Comment:
Why this condition `!settings.inline` added ?

Reply:
This is also used in the `change` variant. It's needed because of #1611 .
Without that, the error message is not getting updated after revalidation.
That's also the main reason why this issue fixes the other 2 SUI issues

Reply:
And the removement of inline errors is already done inside the `validate.field` method, that's why the code first checks if the form should have inline errors